src/lib/Behat/BusinessContext/AdministrationContext.php

```diff
@@ -31,38 +35,129 @@ public function iSeeList(string $pageName): void
+     * @param string $newItemType
      */
-    public function iStartCreatingNew(string $newItemKind): void
+    public function iStartCreatingNew(string $newItemType): void
     {
-        if (!\array_key_exists($newItemKind, $this->itemMapping)) {
-            throw new \InvalidArgumentException(sprintf('Unrecognised item type name: %s', $newItemKind));
+        if (!\array_key_exists($newItemType, $this->itemCreateMapping)) {
+            throw new \InvalidArgumentException(sprintf('Unrecognised item type name: %s', $newItemType));
         }
-        PageObjectFactory::createPage($this->utilityContext, $this->itemMapping[$newItemKind])
+        PageObjectFactory::createPage($this->utilityContext, $this->itemCreateMapping[$newItemType])
             ->adminList->clickPlusButton();
     }
 
     /**
      * @Then there's :listElementName on :page list
-     * @Then there's :expected :listElementName on :page list
      */
-    public function isElementOnTheList($listElementName, $page, $expected = true): void
+    public function isElementOnTheList($listElementName, $page): void
     {
-        $msg = '';
-        if ($expected === 'no') {
-            $expected = false;
-            $msg = ' not';
+        $actual = PageObjectFactory::createPage($this->utilityContext, $page)
+            ->adminList->checkListElementExists($listElementName);
+
+        if (!$actual) {
+            throw new ElementNotFoundException(
+                    $this->utilityContext->getSession(),
+                    sprintf("Element \"%s\" is not on the %s list.", $listElementName, $page));
         }
+    }
 
+    /**
+     * @Then there's no :listElementName on :page list
+     */
+    public function isElementNotOnTheList($listElementName, $page): void
+    {
         $actual = PageObjectFactory::createPage($this->utilityContext, $page)
-            ->adminList->checkListElement($listElementName);
+            ->adminList->checkListElementExists($listElementName);
 
-        if ($expected !== $actual) {
+        if ($actual) {
             throw new ElementNotFoundException(
-                    $this->utilityContext->getSession(),
-                    sprintf("Element \"$s\" is$s on the $s list.", $listElementName, $msg, $page));
+                $this->utilityContext->getSession(),
+                sprintf("Element \"%s\" is on the %s list.", $listElementName, $page));
+        }
+    }
+
+    /**
+     *
+     * Check if item is or is not empty, according to $empty param
+     * @param $itemName
+     * @param $page
+     * @param $empty
+     */
+    private function checkContentsStatus($itemName, $page, $empty): void
+    {
+        $contentsCount = PageObjectFactory::createPage($this->utilityContext, $page)
+            ->adminList->checkListItemAttribute($itemName, $this->emptyHeaderMapping[$page]);
+
+        $msg = '';
+        if ($empty){
+            $msg = ' non';
+        }
+
+        if(($contentsCount !== '0') === $empty){
```

Comment:
I'd rename the param to `shouldBeEmpty` (and extract `$contentsCount !== '0'` to a new variable, isEmpty).